### PR TITLE
Coverage for App and Status service

### PR DESCRIPTION
**Pull request by dissoupov**

### CODE REVIEW

File: backend/service/status/status_http.go [hunk 1]
```diff
@@ -37,14 +37,19 @@ func (s *Service) serverStatus() rest.Handle {
 
 func (s *Service) callerStatus() rest.Handle {
 	return func(w http.ResponseWriter, r *http.Request, _ rest.Params) {
+		callerCtx := identity.ForRequest(r)
+		role := callerCtx.Identity().Role()
```
Comment (haykbaluyan) [inline backend/service/status/status_http.go:41]: No nil comparison here? E.g. see in status_grpc.go above.

```
role := "guest"
if callerCtx != nil {
  role = callerCtx.Identity().Role()
}
```

File: backend/trustymain/app.go [hunk 2]
```diff
@@ -241,6 +269,10 @@ func (a *App) loadConfig() error {
 	flags.cpu = app.Flag("cpu", "enable CPU profiling, specify a file to store CPU profiling info").String()
 	flags.isStderr = app.Flag("std", "output logs to stderr").Bool()
 	flags.dryRun = app.Flag("dry-run", "verify config etc, and do not start the service").Bool()
+	flags.hsmCfg = app.Flag("hsm-cfg", "location of the HSM configuration file").String()
+	flags.cryptoProvs = app.Flag("crypto-prov", "path to additional Crypto provider configurations").Strings()
+	flags.clientURLs = app.Flag("client-listen-url", "URL for the clients listening end-point").Strings()
```
Comment (haykbaluyan) [inline backend/trustymain/app.go:274]: it is a bit unclear. Can this be plural URLs or just a single URL? The variable name is plural while flag name is singular.

File: backend/trustymain/app.go [hunk 3]
```diff
@@ -241,6 +269,10 @@ func (a *App) loadConfig() error {
...
+	flags.healthURLs = app.Flag("health-listen-url", "URL for the health listening end-point").Strings()
```
Comment (haykbaluyan) [inline backend/trustymain/app.go:275]: Same here. The variable name is plural while the flag name is singular.

File: tests/testutils/testutils.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+	"github.com/go-phorce/dolly/algorithms/guid"
+)
+
+var (
+	nextPort    = int32(0)
+	testDirPath = filepath.Join(os.TempDir(), "tests", "trusty", guid.MustCreate())
+)
+
+// CreateURLs returns URL with a random port
+func CreateURLs(scheme, host string) string {
```
Comment (haykbaluyan) [inline tests/testutils/testutils.go:19]: Needs a unit test. Maybe later. I understand it used rand but at least we can have a unit test that for example checks that the returned string starts with `schema://host` and that the port is within an interval used.

**Review by haykbaluyan — APPROVED** [review]

File: tests/testutils/testutils.go (hunk 4 above)
Comment (haykbaluyan) [reply tests/testutils/testutils.go:19]: Ah on the other hand this is for tests only.

File: backend/service/status/status_http.go (hunk 1 above)
Comment (dissoupov) [reply backend/service/status/status_http.go:41]: in HTTP the context is always set and not nil.
In gRPC it's under development still.

File: backend/trustymain/app.go (hunk 2 above)
Comment (dissoupov) [reply backend/trustymain/app.go:274]: it is plural. easier to see whith `trusty --help`
the usage:
```
--client-listen-url=foo --client-listen-url=bar
```

File: backend/trustymain/app.go (hunk 3 above)
Comment (dissoupov) [reply backend/trustymain/app.go:275]: it is plural. easier to see with `trusty --help`
the usage:
```
--client-listen-url=foo --client-listen-url=bar
```

File: tests/testutils/testutils.go (hunk 4 above)
Comment (dissoupov) [reply tests/testutils/testutils.go:19]: this is used in unit tests only. the `tests` folder is excluded from coverage

**Pull request merged**